hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyRegistry.java

```diff
@@ -309,4 +309,17 @@ private void invalidate(DistributedObject distributedObject) {
             ((AbstractDistributedObject) distributedObject).invalidate();
         }
     }
+
+    /**
+     * Force-initializes and publishes all uninitialized proxies in this registry.
+     */
+    void initializeAndPublishProxies() {
+        for (Map.Entry<String, DistributedObjectFuture> entry : proxies.entrySet()) {
+            DistributedObjectFuture future = entry.getValue();
+            if (!future.isSetAndInitialized()) {
+                extractDistributedObject(future);
+                publish(new DistributedObjectEventPacket(CREATED, serviceName, entry.getKey()));
```

Comment:
One thing comes to mind - I see the `ProxyRegistry#extractDistributedObject` method may return `null` in case it wasn't able to retrieve the proxy. Maybe we can skip publication in this case?

Reply:
In `ProxyRegistry.doCreateProxy(..)` , if initialization fails, `future` is removed from `proxies` map and no event is published. 
Additionally, `ProxyRegistry.publish(event)` publishes event only to remote members. There's an additional `eventService.executeEventCallback(callback)` call for local.

Reply:
I have pushed another commit that changes the implementation so that it publishes the event only if proxy initialization succeeded. If proxy initialization failed, the future is removed from the registry, similar to `doCreateProxy()`. I have also added explanation to the method documentation of why only remote events are fired.